capi/src/layout.rs

```diff
@@ -93,6 +93,25 @@ pub extern "C" fn Layout_state(this: &mut Layout, timer: &Timer) -> OwnedLayoutS
     Box::new(this.state(timer))
 }
 
+/// Updates the layout's state based on the timer provided.
+#[no_mangle]
+pub extern "C" fn Layout_update_state(this: &mut Layout, state: &mut LayoutState, timer: &Timer) {
+    this.update_state(state, timer)
+}
+
+/// Updates the layout's state based on the timer provided.
```

Comment:
Should this also say `and encodes it as JSON`?

Also, it might be useful to explain the difference somewhere between the use case for `Layout_update_state_as_json` vs. `Layout_state_as_json`.